chia/full_node/coin_store.py

```diff
@@ -495,3 +496,28 @@ async def _set_spent(self, coin_names: List[bytes32], index: uint32) -> None:
                 raise ValueError(
                     f"Invalid operation to set spent, total updates {rows_updated} expected {len(coin_names)}"
                 )
+
+    # Lookup the most recent unspent lineage that matches a puzzle hash
+    async def get_unspent_lineage_ids_for_puzzle_hash(self, puzzle_hash: bytes32) -> Optional[UnspentLineageIds]:
+        async with self.db_wrapper.reader_no_transaction() as conn:
+            async with conn.execute(
+                "SELECT unspent.coin_name, "
+                "unspent.coin_parent, "
+                "parent.coin_parent "
+                "FROM coin_record AS unspent "
+                "LEFT JOIN coin_record AS parent ON unspent.coin_parent = parent.coin_name "
+                "WHERE unspent.spent_index = 0 "
+                "AND parent.spent_index > 0 "
+                "AND unspent.puzzle_hash = ? "
+                "AND parent.puzzle_hash = unspent.puzzle_hash "
+                "AND parent.amount = unspent.amount",
```

Comment:
this rule was relaxed. the amounts no longer need to match

Reply:
I sent a second commit that adapts to this, now that we relaxed this rule in chia_rs 0.3.3.